jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -115,19 +125,35 @@ protected void doStart() throws Exception
         // The normal strategy obtains the produced task, schedules
         // a new thread to produce more, runs the task and then exits.
         _selectorManager.execute(_strategy::produce);
+    }
 
-        // Set started only if we really are started
-        Start start = new Start();
-        submit(start);
-        start._started.await();
+    protected void onSelectFailed(Throwable cause) throws Exception
+    {
+        LOG.info("Restarting selector: " + toString(), cause);
+        startSelector();
```

Comment:
As discussed, I do not think this should be the default behaviour.
Probably just move the warn done by the caller to the default